PRESUBMIT.py

```diff
@@ -280,6 +280,48 @@ def CheckLicense(input_api, output_api):
     return result
 
 
+def CheckNewSourceFileWithoutGnChangeOnUpload(input_api, output_api):
+    """Checks newly added .cc/.h/.mm files have corresponding GN changes."""
+    files_to_check = (
+        r'.+\.cc$',
+        r'.+\.h$',
+        r'.+\.mm$',
+    )
+
+    def new_sources(f):
+        return input_api.FilterSourceFile(f, files_to_check=files_to_check)
+
+    new_sources = []
+    for f in input_api.AffectedSourceFiles(new_sources):
+        if f.Action() != 'A':
+            continue
+        new_sources.append(f.LocalPath())
+
+    def gn_files(f):
+        return input_api.FilterSourceFile(f, files_to_check=(r'.+\.gn', ))
```

Comment:
this should exclude `chromium_src` dir.

Reply:
Now skipping both for new sources and gn_files.
I believe most of the time new source in chromium_src won't have BUILD.gn changes.
There's a case (maybe just a corner case?) where we might add our own test file that's not for override under chromium_src folder and will need gn changes tho, this PR's current state does not catch that.